proxy/src/lib.rs

```diff
@@ -317,9 +315,19 @@ where
                             ::logging::admin().bg("telemetry").future(telemetry),
                             tap.map_err(|_| {}),
                             metrics_server.map_err(|_| {}),
-                        ).map(|_| {});
+                        );
+
+                    let fut: Box<Future<Item=(), Error=()>> =
+                        if let Some(config_bg) = tls_server_bg {
```

Comment:
... in particular, I think this conditional logic could be replaced with `s/join4/join5/` above if we just created the TLS config changes stream regardless of whether the settings were set.